testsuite/tests/publish/hash-internals/test.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
```

Comment:
I think it would be interesting to test the `alr dev --hash` here as well.

We can test using a tar hosted somewhere and a fake github repo where we put all the corner cases such as symlinks.

Reply:
The constraint is that we wanted to avoid dependencies on external resources for the test suite. Not sure how hard this constraint is, since we need to check out alr subrepos anyway (although all of them come from github).

Reply:
About the fake github repo, this is what the test does currently (it clones the libfoo_git folder, which is a bare repo).